csvqb/csvqb/tests/behaviour/steps/qbwriter.py

```diff
@@ -36,6 +37,37 @@ def step_impl(context, cube_name: str):
     context.cube = _get_single_measure_cube_with_name_and_id(cube_name, None)
 
 
+@Given(
+    'a QbCube named "{cube_name}" with code-list defined in an existing CSV-W "{csvw_file_path}"'
+)
+def step_impl(context, cube_name: str, csvw_file_path: str):
+    tmp_dir = get_context_temp_dir_path(context)
+    csvw_path = tmp_dir / csvw_file_path
+    columns = [
+        QbColumn("A", NewQbDimension.from_data("A code list", _standard_data["A"])),
+        QbColumn(
+            "D",
+            NewQbDimension("D code list", code_list=NewQbCodeListInCsvW(csvw_path)),
+        ),
+        QbColumn(
+            "Value",
+            QbSingleMeasureObservationValue(
+                NewQbMeasure("Some Measure"), NewQbUnit("Some Unit")
+            ),
+        ),
+    ]
+
+    csv_path, _ = get_first_table_schema(csvw_path)
+    code_list_data = pd.read_csv(csvw_path.parent / csv_path)
+    code_list_values = code_list_data["Notation"].sample(3)
+
+    context.cube = Cube(
+        get_standard_catalog_metadata_for_name(cube_name, None),
+        pd.DataFrame({"A": ["a", "b", "c"], "D": code_list_values, "Value": [1, 2, 3]}),
```

Comment:
Because when we set a local code-list against a dimension, we apply foreign key constraints to ensure all of the values used in the dimension are defined in the code-list. These errors are thrown when running csv-lint and I want the test to ensure we pass csv-lint.... so I'm just picking some random codes from the list to make sure it doesn't throw a foreign key validation error.